[BUG] Missing modules whie using with Next.js

### Describe the bug

I am trying to use `useInterval` hook with Next.js App router, and I am getting a error `Module not found: Can't resolve 'lodash.debounce'` while trying to do so.

The Error is specfic to `usehooks-ts 2.11.0`

![image](https://github.com/juliencrn/usehooks-ts/assets/130567419/c617d04f-5a7e-4642-86b5-5cd343e72b8e)


### To Reproduce

1. Create new Next.js App with defaults
2. Install `usehooks-ts 2.11.0`
3. Try to use `useInterval` hook in a client component

### Expected behavior

It should work as it did in previous versions, also the error comes from `useDebounceCallback` which I am not even importing.

### Additional context

_No response_

--- a/.changeset/pink-ravens-fix.md
+++ b/.changeset/pink-ravens-fix.md
@@ -1,0 +1,5 @@
+---
+"usehooks-ts": patch
+---
+
+move `lodash.debounce` to dependencies from peerDependencies

--- a/packages/usehooks-ts/package.json
+++ b/packages/usehooks-ts/package.json
@@ -48,14 +48,11 @@
     "vite": "^5.0.12",
     "vitest": "^1.2.2"
   },
-  "peerDependencies": {
-    "lodash.debounce": "^4",
-    "react": "^16.8.0  || ^17 || ^18"
+  "dependencies": {
+    "lodash.debounce": "^4.0.8"
   },
-  "peerDependenciesMeta": {
-    "lodash.debounce": {
-      "optional": true
-    }
+  "peerDependencies": {
+    "react": "^16.8.0  || ^17 || ^18"
   },
   "engines": {
     "node": ">=16.15.0"

